### PR TITLE
[#941] Allow versions of default plugins to be set

### PR DESCRIPTION
**Pull request by MattWatson-**

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -331,7 +331,7 @@
   String $user                                    = $jenkins::params::user,
   Boolean $manage_group                           = $jenkins::params::manage_group,
   String $group                                   = $jenkins::params::group,
-  Array $default_plugins                          = $jenkins::params::default_plugins,
```
Comment (ekohl) [inline manifests/init.pp]: This makes it a breaking change. Are you sure we want that?

**Comment by MattWatson-** [comment]
#941

File: manifests/init.pp (hunk 1 above)
Comment (MattWatson-) [reply manifests/init.pp]: How would you explicitly set the version for the plugins in your current setup? It seems like with an array you only set the plugins but with the limitation that that you only get the latest

Comment (ekohl) [reply manifests/init.pp]:
```puppet
$my_plugins = {
  'credentials' => { 'version' => 'x.y.z' },
  'structs'     =>  { 'version' => 'a.b.c' },
}
class { 'jenkins': 
  default_plugins => [],
  plugin_hash     => $my_plugins,
}
```

Comment (MattWatson-) [reply manifests/init.pp]: Ah fair enough. That was an oversight on my part and I'll do that instead! If you could put that in the documentation that would be very useful. Will close the issue + PR

**Pull request closed without merging**